sphinxcontrib/confluencebuilder/builder.py

```diff
@@ -885,11 +886,18 @@ def _generate_special_document(self, docname, generator):
                     self.config.confluence_header_file)
                 try:
                     with io.open(fname, encoding='utf-8') as file:
-                        self._cached_header_data = file.read() + '\n'
+                        template_data = file.read() + '\n'
                 except (IOError, OSError) as err:
                     self.warn('error reading file {}: {}'.format(fname, err))
 
-        # cache footer data (if any)
+                # if no data is supplied, the file is plain text
+                if self.config.confluence_header_data is None:
+                    self._cached_header_data = template_data
```

Comment:
There is a potential of `template_data` to be undefined here if the file read attempt results in an exception.

Maybe configure `template_data` to an empty string instead of `self._cached_header_data` at the start of these scopes; or switch it back to just set `self._cached_header_data` and overwrite the value for the template case (either way works for me).

Same would apply for the footer processing section.

Reply:
Done